### PR TITLE
fix: ensure babel-preset-gatsby can be used with unit tests

### LINKED ISSUE #9614: "SyntaxError: Unexpected string" when implementing unit testing guide on stock blog starter
## Description

I am receiving an error when running `yarn test` after installing the [gatsby-starter-blog](https://github.com/gatsbyjs/gatsby-starter-blog) and going through the [Unit testing](https://www.gatsbyjs.org/docs/unit-testing/) guide. The error I receive is:
```
({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,global,jest){import "core-js/modules/es6.date.to-json";
                                                                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    SyntaxError: Unexpected string
```

### Steps to reproduce

I have a demo repo here: [gatsby-starter-blog-jest-error](https://github.com/ebello/gatsby-starter-blog-jest-error)

1. Clone [gatsby-starter-blog-jest-error](https://github.com/ebello/gatsby-starter-blog-jest-error)
2. `yarn`
3. `yarn test`

### Expected result

I should receive the `SyntaxError: Invalid or unexpected token` for `import 'typeface-montserrat'` as referenced in the guide.

### Actual result

I receive an error relating to babel not being able to transform the test file. It seems to be a regression from this commit: 2788f93ee86c710a7a92437e02abfe23decb24ac. If I revert the `babelOptions` back to the following, I get the expected result above.

```
const babelOptions = {
  presets: ["@babel/react", "@babel/env"],
  plugins: [
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-class-properties",
  ],
};
```

### Environment

```
  System:
    OS: macOS High Sierra 10.13.6
    CPU: x64 Intel(R) Core(TM) i7-6567U CPU @ 3.30GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.13.0 - /usr/local/opt/node@10/bin/node
    Yarn: 1.12.1 - /usr/local/bin/yarn
    npm: 6.4.1 - /usr/local/opt/node@10/bin/npm
  Browsers:
    Chrome: 70.0.3538.77
    Firefox: 57.0
    Safari: 12.0
  npmPackages:
    gatsby: ^2.0.19 => 2.0.35 
    gatsby-plugin-feed: ^2.0.8 => 2.0.9 
    gatsby-plugin-google-analytics: ^2.0.5 => 2.0.7 
    gatsby-plugin-manifest: ^2.0.5 => 2.0.7 
    gatsby-plugin-offline: ^2.0.5 => 2.0.10 
    gatsby-plugin-react-helmet: ^3.0.0 => 3.0.1 
    gatsby-plugin-sharp: ^2.0.6 => 2.0.10 
    gatsby-plugin-typography: ^2.2.0 => 2.2.1 
    gatsby-remark-copy-linked-files: ^2.0.5 => 2.0.6 
    gatsby-remark-images: ^2.0.4 => 2.0.5 
    gatsby-remark-prismjs: ^3.0.0 => 3.0.3 
    gatsby-remark-responsive-iframe: ^2.0.5 => 2.0.6 
    gatsby-remark-smartypants: ^2.0.5 => 2.0.6 
    gatsby-source-filesystem: ^2.0.2 => 2.0.6 
    gatsby-transformer-remark: ^2.1.6 => 2.1.11 
    gatsby-transformer-sharp: ^2.1.3 => 2.1.7 
  npmGlobalPackages:
    gatsby-cli: 2.4.4
```

### PR DESCRIPTION
**Pull request by DSchau**
This PR does a few things, namely:

- Fixes a few things with the test stage, e.g. uses commonjs modules for
test running, uses the current node version for @babel/preset-env, etc.
- Fixes up docs slightly for the unit testing guide
- Fixes #9614

### CODE REVIEW

File: docs/docs/unit-testing.md [hunk 1]
```diff
@@ -27,31 +27,26 @@ npm install --save-dev jest babel-jest react-test-renderer identity-obj-proxy 'b
 
 Because Gatsby handles its own Babel configuration, you will need to manually
 tell Jest to use `babel-jest`. The easiest way to do this is to add a `"jest"`
-section in your `package.json`. You can set up some useful defaults at the same
+section in your `jest.config.js`. You can set up some useful defaults at the same
```
Comment (DSchau) [inline docs/docs/unit-testing.md]: Note: I made a few opinionated changes here, namely `package.json` -> `jest.config.js`

File: docs/docs/unit-testing.md [hunk 2]
```diff
@@ -150,11 +145,12 @@ import React from "react"
 import renderer from "react-test-renderer"
 import Bio from "./Bio"
 
-describe("Bio", () =>
+describe("Bio", () => {
```
Comment (DSchau) [inline docs/docs/unit-testing.md]: I think this is quite a bit clearer/real-world, even though both were valid

File: packages/babel-preset-gatsby/src/index.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ function preset(context, options = {}) {
   const stage = process.env.GATSBY_BUILD_STAGE || `test`
 
   if (!targets) {
-    if (stage === `build-html`) {
+    if (stage === `build-html` || stage === `test`) {
```
Comment (DSchau) [inline packages/babel-preset-gatsby/src/index.js]: Think this makes sense, right? Should transpile to whatever the current version of node is in the unit test runner.

File: docs/docs/unit-testing.md (hunk 1 above)
Comment (kakadiadarpan) [inline docs/docs/unit-testing.md]: `jest.config.js` should directly export the defaults, right?

There shouldn't be a `"jest"` section in `jest.config.js`.

Comment (DSchau) [reply docs/docs/unit-testing.md]: Fixed! Thanks a lot!!

**Comment by DSchau** [comment]
Also @kentcdodds if you're feeling spicy 🌶, want to take a look at [`examples/using-jest`](https://github.com/gatsbyjs/gatsby/tree/60d01ab6511922571734d6c8044688fbaf5f7f73/examples/using-jest) and make sure it looks good to your eyes? I used react-testing-library for the bulk of the unit testing functionality, so just want to make sure it largely looks solid!

No worries if not, though :)

File: examples/using-jest/__mocks__/gatsby.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const React = require('react')
+const gatsby = jest.requireActual('gatsby')
+
+module.exports = {
+  ...gatsby,
+  graphql: jest.fn(),
+  Link: jest.fn().mockImplementation(({ to, ...rest }) =>
```
Comment (DSchau) [inline examples/using-jest/__mocks__/gatsby.js]: This is kinda janky, but I also think that this should _probably_ be added to the docs/unit-testing.md document!

Comment (DSchau) [reply examples/using-jest/__mocks__/gatsby.js]: (also just for clarity here, gatsby-link errors out in Jest, so this is a way to side step that issue)

File: examples/using-jest/README.md [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    <img alt="Gatsby" src="https://www.gatsbyjs.org/monogram.svg" width="60" />
+  </a>
+</p>
+<h1 align="center">
+  using Jest
+</h1>
+
+Kick off your next Gatsby app with some great testing practices enabled via [Jest][jest], [react-testing-library][react-testing-library], and of course, [Gatsby][gatsby] 💪
+
+Check out the [unit testing doc][unit-testing-doc] for further info!
```
[the remark this answers is not in the document]
Comment (DSchau) [reply examples/using-jest/README.md]: Yeah! These are markdown style links, so it's like a table of links that it refers to below. Actually a good practice to get into, I think :) Keeps links in one place so they're easy to edit/re-use!

[the remark this answers is not in the document]
Comment (DSchau) [reply examples/using-jest/README.md]: Just as an example if it's helpful!

```markdown
This is a link to [google][google], and [this link][google] will also refer to the google link! 🎉 

[google]: https://google.com
```

[the remark this answers is not in the document]
Comment (shannonbux) [reply examples/using-jest/README.md]: ohhhhhh kewl

[the remark this answers is not in the document]
Comment (shannonbux) [reply examples/using-jest/README.md]: that seems handy. I'll keep that in mind!

**Comment by DSchau** [comment]
@pieh I think we should take a look at this today. Wanna give it a review? 👀

File: docs/docs/unit-testing.md [hunk 6]
```diff
@@ -318,3 +283,8 @@ though remember you may need to install the Babel 7 versions. See
 
 For more information on Jest testing, visit
 [the Jest site](https://jestjs.io/docs/en/getting-started).
+
+For an example encapsulating all of these techniques--and a full unit test suite with [react-testing-library][react-testing-library], check out the [using-jest][using-jest] example.
+
+[using-jest]: examples/using-jest
```
Comment (pieh) [inline docs/docs/unit-testing.md]:
```suggestion
[using-jest]: https://github.com/gatsbyjs/gatsby/tree/master/examples/using-jest
```
Need to be absolute as we will be linking from gatsbyjs.org

Comment (DSchau) [reply docs/docs/unit-testing.md]: Thanks @pieh!

**Comment by DSchau** [comment]
@kentcdodds hey sure, thanks for chiming in! I'll update--I appreciate the feedback!

**Comment by DSchau** [comment]
```
Successfully published:
 - babel-preset-gatsby@0.1.3
```

**Pull request merged**